sdk/monitor/monitor-query/test/public/shared/testShared.ts

```diff
@@ -27,67 +31,53 @@ export interface RecorderAndMetricsClient {
   recorder: Recorder;
 }
 
-export const testEnv = new Proxy(replaceableVariables, {
+export const testEnv = new Proxy(envSetupForPlayback, {
   get: (target, key: string) => {
     return env[key] || target[key];
   },
 });
 
-export const environmentSetup: RecorderEnvironmentSetup = {
-  // == Recorder Environment Setup == Add the replaceable variables from
-  // above
-  replaceableVariables,
-
-  // We don't use this in the template, but if we had any query parameters
-  // we wished to discard, we could add them here
-  queryParametersToSkip: [],
-
-  // Finally, we need to remove the AAD `access_token` from any requests.
-  // This is very important, as it cannot be removed using environment
-  // variable or query parameter replacement.  The
-  // `customizationsOnRecordings` field allows us to make arbitrary
-  // replacements within recordings.
-  customizationsOnRecordings: [
-    (recording: string): any =>
-      recording.replace(/"access_token":"[^"]*"/g, `"access_token":"access_token"`),
-  ],
-};
+// export function createRecorderAndMetricsClient(context: Context): RecorderAndMetricsClient {
+//   const recorder = record(context, environmentSetup);
+//   return {
+//     client: new MetricsQueryClient(createTestClientSecretCredential()),
+//     recorder,
+//   };
+// }
+
+export async function createRecorderAndMetricsClient(
+  recorder: Recorder
+): Promise<RecorderAndMetricsClient> {
+  //const recorder = record(context, environmentSetup);
+  await recorder.start(recorderOptions);
 
-export function createRecorderAndMetricsClient(context: Context): RecorderAndMetricsClient {
-  const recorder = record(context, environmentSetup);
+  const client = new MetricsQueryClient(createTestCredential());
+
+  recorder.configureClient(client["_metricsClient"]);
   return {
-    client: new MetricsQueryClient(createTestClientSecretCredential()),
-    recorder,
+    client: client,
+    recorder: recorder,
   };
 }
 
-export function createRecorderAndLogsClient(
-  context: Context,
+export async function createRecorderAndLogsClient(
+  recorder: Recorder,
   retryOptions?: ExponentialRetryPolicyOptions
-): RecorderAndLogsClient {
-  const recorder = record(context, environmentSetup);
+): Promise<RecorderAndLogsClient> {
+  await recorder.start(recorderOptions);
+
+  const client = new LogsQueryClient(createTestCredential(), {
+    retryOptions,
+  });
+
+  recorder.configureClient(client["_logAnalytics"]);
```

Comment:
Now that the pipeline customization API changes are in, we should be able to do

```ts
const client = new LogsQueryClient(createTestCredential, recorder.configureClientOptions({ retryOptions });
```

instead of the private field access workaround we were relying on before!

Reply:
Cool, thanks for pointing this out @timovv 👍🏼